{
    "active_lock_reason": null,
    "assignee": null,
    "assignees": [],
    "author_association": "NONE",
    "body": "### Neovim version (nvim -v)\n\nNVIM v0.8.1 Build type: Release\n\n### Vim (not Nvim) behaves the same?\n\nVim doesn't have built in LSP support. With no LSP clients attached, both behave the same (vim 9.0 patches 1-813)\n\n### Operating system/version\n\n6.0.8-arch x86_64 GNU/Linux\n\n### Terminal name/version\n\nkitty 0.26.5\n\n### $TERM environment variable\n\nxterm-kitty\n\n### Installation\n\nArch repositories\n\n### How to reproduce the issue\n\nThe file I am getting crashes with can be obtained from `curl https://blogsurf.io/rankings/2`. It's 'only' 3.5 MB, but it's one long line.\r\n\r\nI can get the crash to occur with the following config:\r\n```\r\nreturn require(\"packer\").startup(function()\r\n    use({ \"neovim/nvim-lspconfig\" })\r\n    use({\r\n        \"jose-elias-alvarez/null-ls.nvim\",\r\n        config = function()\r\n            require(\"plugins.null_ls\")\r\n        end,\r\n        requires = { \"nvim-lua/plenary.nvim\" },\r\n    })\r\nend)\r\nlocal null_ls = require(\"null-ls\")\r\nnull_ls.setup({\r\n    sources = {\r\n        null_ls.builtins.diagnostics.todo_comments,\r\n        null_ls.builtins.hover.printenv,\r\n        null_ls.builtins.diagnostics.codespell\r\n    },\r\n})\r\n```\r\n\r\nI then open the file and press `$` - nvim hangs for a few seconds and then I am logged out. I get the following log output:\r\n```\r\nNov 16 21:46:31 archlaptop kernel: nvim invoked oom-killer: gfp_mask=0x140dca(GFP_HIGHUSER_MOVABLE|__GFP_COMP|__GFP_ZERO), order=0, oom_score_adj=0\r\nNov 16 21:46:31 archlaptop kernel: CPU: 8 PID: 2082 Comm: nvim Not tainted 6.0.8-arch1-1 #1 9d710c2a17e975eb42321436dd1a075089abf66c\r\nNov 16 21:46:31 archlaptop kernel: Call Trace:\r\nNov 16 21:46:31 archlaptop kernel:  <TASK>\r\nNov 16 21:46:31 archlaptop kernel:  dump_stack_lvl+0x48/0x60\r\nNov 16 21:46:31 archlaptop kernel:  dump_header+0x4a/0x1ff\r\nNov 16 21:46:31 archlaptop kernel:  oom_kill_process.cold+0xb/0x10\r\nNov 16 21:46:31 archlaptop kernel:  out_of_memory+0x27e/0x520\r\nNov 16 21:46:31 archlaptop kernel:  __alloc_pages_slowpath.constprop.0+0xcbe/0xe10\r\nNov 16 21:46:31 archlaptop kernel:  __alloc_pages+0x224/0x250\r\nNov 16 21:46:31 archlaptop kernel:  __folio_alloc+0x1b/0x50\r\nNov 16 21:46:31 archlaptop kernel:  vma_alloc_folio+0x8c/0x300\r\nNov 16 21:46:31 archlaptop kernel:  __handle_mm_fault+0x933/0xf70\r\nNov 16 21:46:31 archlaptop kernel:  handle_mm_fault+0xb2/0x290\r\nNov 16 21:46:31 archlaptop kernel:  do_user_addr_fault+0x1be/0x6a0\r\nNov 16 21:46:31 archlaptop kernel:  exc_page_fault+0x74/0x170\r\nNov 16 21:46:31 archlaptop kernel:  asm_exc_page_fault+0x26/0x30\r\nNov 16 21:46:31 archlaptop kernel: RIP: 0033:0x7ff1764a7e07\r\nNov 16 21:46:31 archlaptop kernel: Code: Unable to access opcode bytes at RIP 0x7ff1764a7ddd.\r\nNov 16 21:46:31 archlaptop kernel: RSP: 002b:00007fff6a9ec1f8 EFLAGS: 00010203\r\nNov 16 21:46:31 archlaptop kernel: RAX: 00007fedea827028 RBX: 00007fedea827010 RCX: 0000000000000020\r\nNov 16 21:46:31 archlaptop kernel: RDX: 00000000000007ac RSI: 00007fef5bd900a8 RDI: 00007fedeb4cb040\r\nNov 16 21:46:31 archlaptop kernel: RBP: 0000000022532d3a R08: ffffffffffffffe8 R09: 0000000000000000\r\nNov 16 21:46:31 archlaptop kernel: R10: 00000000000009b3 R11: 0000000000300000 R12: 00000000033707c4\r\nNov 16 21:46:31 archlaptop kernel: R13: 00007ff17632f380 R14: 00000000033707c4 R15: 00007ff17632f3e0\r\nNov 16 21:46:31 archlaptop kernel:  </TASK>\r\n```\n\n### Expected behavior\n\nDon't run out of memory, since `nvim -u NONE` handles the file just fine.\n\n### Actual behavior\n\nThe template asks me to reproduce with nvim --clean, but given the issue only turns up with a language server attached, I cannot do this. If desired, I could definitely try to reproduce without using plugins, but I think some minimal configuration is necessary, and I'm not sure what that would be for this case. Maybe test other files with similarly long lines (I think this one was over 350000 characters long) with an appropriate language server which can handle that?\r\n\r\nOr, let me know if I should report with the null-ls plugin instead. I wasn't sure where would be best, since it's neovim itself which is able to trigger the oom killer.",
    "closed_at": null,
    "closed_by": null,
    "comment_data": [
        {
            "author_association": "MEMBER",
            "body": "It is better not to add any third-party plugins in the minimal configuration . ",
            "created_at": "2022-11-17T12:28:23Z",
            "html_url": "https://github.com/neovim/neovim/issues/21084#issuecomment-1318565681",
            "id": 1318565681,
            "issue_url": "https://api.github.com/repos/neovim/neovim/issues/21084",
            "node_id": "IC_kwDOAPphoM5Ol7cx",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 0,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 0,
                "url": "https://api.github.com/repos/neovim/neovim/issues/comments/1318565681/reactions"
            },
            "updated_at": "2022-11-17T12:28:23Z",
            "url": "https://api.github.com/repos/neovim/neovim/issues/comments/1318565681",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/41671631?v=4",
                "events_url": "https://api.github.com/users/glepnir/events{/privacy}",
                "followers_url": "https://api.github.com/users/glepnir/followers",
                "following_url": "https://api.github.com/users/glepnir/following{/other_user}",
                "gists_url": "https://api.github.com/users/glepnir/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/glepnir",
                "id": 41671631,
                "login": "glepnir",
                "node_id": "MDQ6VXNlcjQxNjcxNjMx",
                "organizations_url": "https://api.github.com/users/glepnir/orgs",
                "received_events_url": "https://api.github.com/users/glepnir/received_events",
                "repos_url": "https://api.github.com/users/glepnir/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/glepnir/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/glepnir/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/glepnir",
                "user_view_type": "public"
            }
        }
    ],
    "comments": 1,
    "comments_url": "https://api.github.com/repos/neovim/neovim/issues/21084/comments",
    "created_at": "2022-11-17T04:27:12Z",
    "events_url": "https://api.github.com/repos/neovim/neovim/issues/21084/events",
    "html_url": "https://github.com/neovim/neovim/issues/21084",
    "id": 1452679022,
    "labels": [
        {
            "color": "f9d0c4",
            "default": true,
            "description": "issues reporting wrong behavior",
            "id": 77997474,
            "name": "bug",
            "node_id": "MDU6TGFiZWw3Nzk5NzQ3NA==",
            "url": "https://api.github.com/repos/neovim/neovim/labels/bug"
        },
        {
            "color": "c5def5",
            "default": false,
            "description": null,
            "id": 662566370,
            "name": "lsp",
            "node_id": "MDU6TGFiZWw2NjI1NjYzNzA=",
            "url": "https://api.github.com/repos/neovim/neovim/labels/lsp"
        }
    ],
    "labels_url": "https://api.github.com/repos/neovim/neovim/issues/21084/labels{/name}",
    "locked": false,
    "milestone": null,
    "node_id": "I_kwDOAPphoM5Wlh9u",
    "number": 21084,
    "performed_via_github_app": null,
    "reactions": {
        "+1": 0,
        "-1": 0,
        "confused": 0,
        "eyes": 0,
        "heart": 0,
        "hooray": 0,
        "laugh": 0,
        "rocket": 0,
        "total_count": 0,
        "url": "https://api.github.com/repos/neovim/neovim/issues/21084/reactions"
    },
    "repository_url": "https://api.github.com/repos/neovim/neovim",
    "state": "open",
    "state_reason": null,
    "sub_issues_summary": {
        "completed": 0,
        "percent_completed": 0,
        "total": 0
    },
    "timeline_url": "https://api.github.com/repos/neovim/neovim/issues/21084/timeline",
    "title": "Opening file with very long line with LSP enabled causes out-of-memory crash",
    "updated_at": "2025-01-19T11:37:23Z",
    "url": "https://api.github.com/repos/neovim/neovim/issues/21084",
    "user": {
        "avatar_url": "https://avatars.githubusercontent.com/u/4873173?v=4",
        "events_url": "https://api.github.com/users/dryya/events{/privacy}",
        "followers_url": "https://api.github.com/users/dryya/followers",
        "following_url": "https://api.github.com/users/dryya/following{/other_user}",
        "gists_url": "https://api.github.com/users/dryya/gists{/gist_id}",
        "gravatar_id": "",
        "html_url": "https://github.com/dryya",
        "id": 4873173,
        "login": "dryya",
        "node_id": "MDQ6VXNlcjQ4NzMxNzM=",
        "organizations_url": "https://api.github.com/users/dryya/orgs",
        "received_events_url": "https://api.github.com/users/dryya/received_events",
        "repos_url": "https://api.github.com/users/dryya/repos",
        "site_admin": false,
        "starred_url": "https://api.github.com/users/dryya/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/dryya/subscriptions",
        "type": "User",
        "url": "https://api.github.com/users/dryya",
        "user_view_type": "public"
    }
}